Documentation/UsingSetting/AddTypoScriptWithExtensions.rst

```diff
@@ -6,60 +6,167 @@
 Add TypoScript in your extension
 ================================
 
+..  versionchanged:: 13.1
+    TypoScript on a per site basis can now be included via
+    :ref:`sites and sets <typoscript-site-sets>`.
+
 .. note::
 
    This part is written for extension developers.
 
 
 .. index:: TypoScript in extensions; File locations
 
+.. _extdev-add-typoscript-extension:
+
 Create TypoScript files in your extension
 =========================================
 
-TypoScript files should have the ending :file:`.typoscript`.
+TypoScript files **must** have the ending :file:`.typoscript`.
 
-They are located in :file:`Configuration/TypoScript` within your
-extension.
+They are located in :path:`Configuration/Sets/MySet` within your
+extension. Read more about how to
+:ref:`provide the TypoScript as set for TYPO3 v13 and above <extdev-add-typoscript-sets>`
+and :ref:`how to provide TypoScript for both TYPO3 v13 and v12 <extdev-add-typoscript-sets-v12>`.
 
-Usually, you will have the following structure:
+*   :file:`constants.typoscript` contains the constants
+*   :file:`setup.typoscript` contains the TypoScript setup
 
-.. code-block:: none
 
-   Configuration/
-   └── TypoScript
-       ├── constants.typoscript
-       └── setup.typoscript
+.. _extdev-add-typoscript-sets:
 
+TypoScript provided as site set (only TYPO3 v13.1 and above)
+============================================================
 
-* :file:`constants.typoscript` contains the constants
-* :file:`setup.typoscript` contains the TypoScript setup
+The file structure of the extension could, for example look like this:
 
+..  directory-tree::
+    :show-file-icons: true
+
+    *   Classes
+
+        *   ...
+
+    *   Configuration
+
+        *   Sets
+
+            *   MyExtension
+
+                *   :ref:`config.yaml <extdev-add-typoscript-sets-main>`
+                *   :ref:`constants.typoscript <extdev-add-typoscript-sets-typoscript>`
+                *   :ref:`setup.typoscript <extdev-add-typoscript-sets-typoscript>`
+
+            *   MyExtensionWithACoolFeature
+
+                *   :ref:`config.yaml <extdev-add-typoscript-sets-feature>`
+                *   :ref:`setup.typoscript <extdev-add-typoscript-sets-typoscript>`
+
+    *   Resources
+
+        *   ...
+
+    *   composer.json
+    *   ...
+
+With the extensions TypoScript residing in :file:`EXT:my_extension/Configuration/TypoScript`
```

Comment:
```suggestion
With the extension's TypoScript residing in :file:`EXT:my_extension/Configuration/TypoScript`
```

You mean in Configuration/Sets/MySet though!? The description above already shows a file structure for Sets. Is the intention here how to move from Configuration/ to Sets?

I don't really understand (and maybe I'm a good target audience, since TypoScript and Sets are complicated to me)